test/server/guarddog_impl_test.cc

```diff
@@ -189,6 +189,48 @@ TEST_F(GuardDogMissTest, MegaMissTest) {
   unpet_dog = nullptr;
 }
 
+TEST_F(GuardDogMissTest, MissCountTest) {
+  // This tests a flake discovered in the MissTest where real timeout or
+  // spurious condition_variable wakeup causes the counter to get incremented
+  // more than it should be.
+  ON_CALL(time_source_, currentTime())
+      .WillByDefault(testing::Invoke([&]() {
```

Comment:
Sure, I can make this change (and the other spots where this idiom is repeated).